dvc/repo/metrics/show.py

```diff
@@ -88,21 +93,30 @@ def show(
     all_commits=False,
 ):
     res = {}
+    metrics_found = False
 
     for rev in repo.brancher(
         revs=revs,
         all_branches=all_branches,
         all_tags=all_tags,
         all_commits=all_commits,
     ):
+
         metrics = _collect_metrics(repo, targets, recursive)
+
+        if not metrics_found and metrics:
+            metrics_found = True
+
         vals = _read_metrics(repo, metrics, rev)
 
         if vals:
             res[rev] = vals
 
     if not res:
-        raise NoMetricsError()
+        if metrics_found:
+            raise NoMetricsParsedError()
+        else:
+            raise NoMetricsError()
```

Comment:
Let's not create a yet another exception type here, since it is really the same error with just a tiny difference. Let's just raise with an explicit message,